### PR TITLE
Template Part Block: Use buttons for area selection

### LINKED ISSUE #29295: Template part: Area selection UI
When #29159 is merged it will be possible to update the `area` of the selected Template Part:

Screenshot 2021-02-24 at 10 39 02

Since we have dedicated icons for these areas that will be surfaced to users in the Toolbar, List View, and the Inspector, it may be worth exploring designs that incorporate them into this selection flow in order to establish holistic continuity in the UI.

One option might be to use a segmented control instead of a dropdown:

Another option would be to stick with the select but enrich the dropdown:

Screenshot 2021-02-24 at 11 04 13

Whichever solution we go with, we should be aware that this element will likely exist in other environments, e.g. #29222.

### PR DESCRIPTION
**Pull request by creativecoder**
## Description

Resolves #29295

## How has this been tested?

1. Open the site editor
2. Select a template part block, such as the header (the list view is an easy way to do this).
3. Open the Advanced section of the sidebar options and choose the area.

## Screenshots 

Template Part Block > Advanced settings in sidebar

image

image

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by jameskoster** [comment]
Looking good! :) 

I updated the icon / description for the general area.

Screenshot 2021-03-19 at 09 24 33

The width of these tooltips makes them a little awkward to read. Is it possible to wrap them?

**Comment by creativecoder** [comment]
> I updated the icon / description for the general area.

Thank you!

> The width of these tooltips makes them a little awkward to read. Is it possible to wrap them?

Yes! Now updated.

I'm a little unsure about the screen reader accessibility of using tooltips to describe the template part areas. I modified the Tooltip component to be able to receive an id and be the target of aria-describedby (declared on the buttons)--this seems to help when testing with VoiceOver, but I'd want a more experienced opinion before merging this.

File: packages/components/src/tooltip/index.js
Comment (david-szabo97) [inline packages/components/src/tooltip/index.js:100]: Hmmmm 🤔 This seems to be a rather big modification. I wonder if we should use an explicit `popoverProps` property instead of `...` rest. We need to update the README too.

As a last thought, maybe we should move this to a separate PR?

**Comment by david-szabo97** [comment]
It's working as expected. My only concern is the modification of the `Tooltip` component. The changes make sense but it feels it should be its own PR. Looking forward to others' thoughts.

https://user-images.githubusercontent.com/2256104/111965470-0ecf8100-8af6-11eb-96cb-558d0531462d.mov

**Comment by jameskoster** [comment]
+1 to moving the a11y discussion to a separate issue since it will effectively relate to all tooltip usage across the app.

**Comment by david-szabo97** [comment]
@creativecoder Could you break this into multiple PRs?

We should remove the tooltip changes from this PR and create a separate PR for just the tooltip. Then we can create a follow-up PR to add the accessibility parts once we have the tooltip PR merged. You can use the tooltip PR as a base for the follow-up PR so we can show a use-case as well in the tooltip PR.

So basically we will have 3 PRs in the end:
1. This PR which introduces the buttons for area selection (without a11y in mind)
2. Tooltip PR which adds the new prop to the `Tooltip` component so we can passthrough props to the `Popover` component
3. A accessibility follow-up PR to the 1. one, using 2. as the base branch

**Comment by creativecoder** [comment]
> Could you break this into multiple PRs?

Yes! Will work on that soon. I'd like to leave the className passthrough (Tooltip prop that's passed to the Popover component) so we can control the tooltip width. That seems less invasive, and therefore okay? I'll be sure to update the component readme, accordingly.

**Comment by david-szabo97** [comment]
> I'd like to leave the className passthrough (Tooltip prop that's passed to the Popover component) so we can control the tooltip width. That seems less invasive, and therefore okay?

I would be still concerned about the PR's responsibility. In my opinion, it's better to avoid making core changes in a mostly Site Editor related PR. Smaller PRs and single responsibility PRs are easier to revert and discuss.

File: packages/block-library/src/template-part/edit/advanced-controls.js
Comment (Addison-Stavlo) [inline packages/block-library/src/template-part/edit/advanced-controls.js:81]: I wonder if `RadioGroup`/`Radio` [components](https://developer.wordpress.org/block-editor/reference-guides/components/radio-group/) would be able to fit well and be adaptable for this case?  It seems pretty similar visually and behaviorally to what we are going for.

File: packages/block-library/src/template-part/edit/advanced-controls.js
Comment (Addison-Stavlo) [inline packages/block-library/src/template-part/edit/advanced-controls.js:28]: Not something that needs changed in this PR, but we should centralize these descriptions/definitions so we don't need to re-define them in a handful of places.  I'm trying that as part of https://github.com/WordPress/gutenberg/pull/30395, so a follow up once these both land would be to update this to get that data from the core editor store.

**Review by Addison-Stavlo — COMMENTED** [review]
Just some minor thoughts, its lookin pretty good so far. 😁

**Review by jameskoster — APPROVED** [review]
Approving the design. If the width has to be fixed in a follow-up PR, so be it :D

File: packages/block-library/src/template-part/edit/advanced-controls.js
Comment (david-szabo97) [reply packages/block-library/src/template-part/edit/advanced-controls.js:81]: +1 for RadioGroup and Radio components. They seem to fit this particular situation.

Comment (creativecoder) [reply packages/block-library/src/template-part/edit/advanced-controls.js:81]: Thanks for the idea, I will take a look.

**Comment by david-szabo97** [comment]
👋 Would love to see this merged 😄 Could you rebase it? Happy to approve once tests are passing and rebased. We can create a follow-up PR for the [a11y](https://github.com/WordPress/gutenberg/pull/30005#discussion_r607144676) comment.

**Comment by creativecoder** [comment]
Closing in favor of https://github.com/WordPress/gutenberg/pull/45764, which uses the newer ToggleGroupControl component.

**Pull request closed without merging**
